src/sipmpdf/functions.py

```diff
@@ -223,25 +237,56 @@ def darkcurrent_response(x: np.float64,
   eps = gain * resolution
   lo = pedestal
   up = pedestal + gain
+  
   return kern.where((x > (lo + eps)) & (x < (up - eps)),
-                    ((1 / (x - lo)) + (1 / (up - x))) / (2 * kern.log(
-                      (up - lo - eps) / eps)), 0)
+                    ((1 / (x - lo)) + (1 / (up - x))) / (2 * kern.log((up - lo - eps) / eps)), 0)
 
 
+##This part needs work, uncelar if it works correctly
 def ap_response_smeared(x: np.float64, smear: np.float64, n_ap: np.int64,
-                        beta: np.float64) -> np.float64:
-  kern = kernel_switch(x, smear, n_np, beta)
-  pass
-
+                        beta: np.float64,pedestal: np.float64, 
+                        gain: np.float64,total: np.int64) -> np.float64:
+  kern = kernel_switch(x, smear, n_ap, beta, pedestal, gain, total)
+  
+  ##did this based on the paper which says this only matters when i<2
+  if n_ap>1:
+    return ap_response(x=x,n_ap=n_ap,beta=beta, pedestal=pedestal,gain=gain,total=total)
+  else:
+    pk=pedestal+total*gain
+    import math
+    from scipy import special
+    result=kern.exp((pk-x)/beta)*(1-special.erf((x-pk)/(smear*math.sqrt(2))))/(2*beta)##Not sure about if this will work with tensorflow
```

Comment:
`scipy.special` will not be availalbe to tensorflow. You can use the `normal_cdf` function implemented in the same file.

Reply:
Also use `kern.sqrt`